go/tools/builders/cover.go

```diff
@@ -151,14 +151,18 @@ func addNamedImport(fset *token.FileSet, f *ast.File, name, path string) {
 	// Our new import, preceded by a blank line,  goes after the package declaration
 	// and after the comment, if any, that starts on the same line as the
 	// package declaration.
-	impDecl.TokPos = f.Package
+	impDecl.TokPos = f.Name.End() + 2 // +2 for a blank line
 	file := fset.File(f.Package)
 	pkgLine := file.Line(f.Package)
+	importLength := impDecl.End() - impDecl.Pos() + 1
 	for _, c := range f.Comments {
-		if file.Line(c.Pos()) > pkgLine {
-			break
+		if file.Line(c.Pos()) <= pkgLine {
```

Comment:
I don't think I understand what's happening here to be sure whether it's correct.

It looks like there's a function in `x/tools` that does this: [`astutil.AddNamedImport`](https://github.com/golang/tools/blob/d00afeaade8f1e68fb815705aa42d704c1b6df35/go/ast/astutil/imports.go#L28). Instead of trying to fix this, let's copy that function here, along with a comment pointing to the original implementation.